lua/entities/gmod_wire_expression2/base/parser.lua

```diff
@@ -593,6 +593,12 @@ function Parser:Stmt10()
 					self:Error("Function name must appear after colon (:)")
 				end
 			end
+
+			if not Name then self:Error("Function name must follow function declaration") end
```

Comment:
This check and the ones below it should be moved out of this scope, otherwise this (`function ()`) for example will give you a nil error down below because this part is never ran.

Reply:
Oof, good spot!